### PR TITLE
statistics: add gc in hot peer cache

### LINKED ISSUE #8698: Region may not be removed in hot peer cache after merging
## Bug Report

If the region is not completely cooled down and the merge happens, it might be left in the hot peer cache and never be cleaned up. The merged region will never send a heartbeat again, resulting in checkExpiredTask not working.

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #8698

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/statistics/hot_peer_cache.go [hunk 1]
```diff
@@ -115,6 +116,7 @@ func (f *HotPeerCache) UpdateStat(item *HotPeerStat) {
 		return
 	}
 	f.incMetrics(item.actionType, item.StoreID)
+	f.removeExpiredItems()
```
Comment (rleungx) [inline pkg/statistics/hot_peer_cache.go:119]: Will it be called every time we receive a region heartbeat?

Comment (lhy1024) [reply pkg/statistics/hot_peer_cache.go:119]: L554-L556

File: pkg/statistics/hot_cache.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ func (w *HotCache) CheckReadAsync(task func(cache *HotPeerCache)) bool {
 func (w *HotCache) RegionStats(kind utils.RWType, minHotDegree int) map[uint64][]*HotPeerStat {
 	ret := make(chan map[uint64][]*HotPeerStat, 1)
 	collectRegionStatsTask := func(cache *HotPeerCache) {
-		ret <- cache.RegionStats(minHotDegree)
+		ret <- cache.PeerStats(minHotDegree)
```
Comment (rleungx) [inline pkg/statistics/hot_cache.go:83]: Why change the name, it's not consistent with other places.

Comment (lhy1024) [reply pkg/statistics/hot_cache.go:83]: Okay, I'll keep the TODO and change them in another PR uniform.

**Review by rleungx — APPROVED** [review]

**Review by okJiang — APPROVED** [review]

**Pull request merged**